yt/visualization/profile_plotter.py

```diff
@@ -982,6 +983,45 @@ def _initialize_instance(
         obj._initfinished = True
         return obj
 
+    @invalidate_data
+    def _switch_ds(self, new_ds, data_source=None):
+        old_object = self.data_source
+        name = old_object._type_name
+        kwargs = {n: getattr(old_object, n) for n in old_object._con_args}
+        kwargs["center"] = getattr(old_object, "center", None)
+        if data_source is not None:
+            if name != "proj":
+                raise RuntimeError(
+                    "The data_source keyword argument "
+                    "is only defined for projections."
+                )
+            kwargs["data_source"] = data_source
```

Comment:
this can probably be dropped completely, but I'll need a hand to confirm that I'm reading this right.